prov/gni/src/gnix_ep.c

```diff
@@ -409,18 +409,25 @@ int gnix_ep_open(struct fid_domain *domain, struct fi_info *info,
 	ep_priv->rx_progress_fn = NULL;
 
 	/*
-	 * TODO: hookup a nic to this ep, may add more later if
-	 *  fi_tx_context, etc. is invoked on this endpoing
+	 * If we've maxed out the number of nics for this domain/ptag,
+	 * search the list of existing nics
 	 */
 
-        list_for_each(&gnix_nic_list,elem,list) {
-		if ((elem->ptag == domain_priv->ptag) &&
-			(elem->cookie == domain_priv->cookie)
-			&& (elem->cdm_id == domain_priv->cdm_id)) {
-			nic = elem;
-			atomic_inc(&nic->ref_cnt);
-			break;
+	pthread_mutex_lock(&gnix_nic_list_lock);
+
+	if (gnix_nics_per_ptag[domain_priv->ptag] ==
```

Comment:
good question.  The answer is yes.  The aries hw only has 8 bits for ptag.  Look in gni_pub.h
grep for GNI_PTAG_LND (lustre's ptag) and then the rest.  max user ptag  is GNI_PTAG_USER_END.